### ISSUE TITLE
$(...).pushpin is not a function

### ISSUE THREAD

**Issue by jessor** [Dogfalo/materialize#2849]
I'm initializing Pushpin in a WordPress theme like this:
```
(function($){
  $(function(){
    $(".button-collapse").sideNav();  // works
    $('#tribe-events-bar').pushpin(); // error: $(...).pushpin is not a function
   });
})(jQuery);  
```

Removing the wrapping `$(document).ready(function() {` `});` in materialize.js fixes this.

**Comment by Kail0** [comment]
Well, you can try 
`jQuery(document).ready(function(){` ` });` instead of `$(document).ready(function() { });`

**Comment by gnicol-bzh** [comment]
The problem seems to be the (not needed) `$(document).ready(function() { ... });` inside `pushpin.js`.

CF [$(…).pushpin is not a function - reactJs, Materializecss](http://stackoverflow.com/questions/31091454/pushpin-is-not-a-function-reactjs-materializecss)

**Comment by Kail0** [comment]
I'm using pushpin in my own WP theme and I'm not affected by this issue..

**Comment by gnicol-bzh** [comment]
I'm having the issue with just:
```javascript
(function($) {
    'use strict';
   
    $('.side-menu-wrapper').pushpin({ 
        top: $('.side-menu-wrapper').offset().top 
    });

    //[other js init...]

})(jQuery);
```
in my `main.js` file.

And it works well when I remove the `ready()` in `pushpin.js`. 

As far as I'm concerned, I don't understand why the whole function is wrapped in a `ready` event, especially when wrapped in a `(function ($) { }( jQuery ))`. It won't work if this `ready` event is already fired when `pushpin.js`/`materialize.js` is loaded. Only the init part of the script should be wrapped in a `ready` or `load` event.

**Comment by acburst** [comment]
Yes this was definitely a bug on our part, not sure how that happened. 

Fixed in bc257335f5259966125444e78e624b7b71a4c78b

Thanks for pointing this out
